lib/options.nix

```
@@ -391,6 +391,39 @@ rec {
       );
     };
 
+  mkOptionalPackage =
+    pkgs: name: package:
+    if lib.isOption package then
+      package
+    else if package != null then
+      lib.mkPackageOption pkgs name {
+        nullable = true;
+        default = package;
+      }
+    else
+      (lib.mkOption {
+        type = types.nullOr types.package;
+        description = ''
+          Package to use for ${name}.
+        '';
+      })
+      // {
+        _unpackaged = true;
+        _packageName = name;
+      };
+
+  useOptionalPackage =
+    options: config: opt:
```

```suggestion
    options: config: loc:
```

or

```suggestion
    options: config: path:
```

(not sure we really need `config` either, because we have `opt.value` available)